dbms/src/WindowFunctions/IWindowFunction.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <DataStreams/WindowBlockInputStream.h>
+#include <DataTypes/getLeastSupertype.h>
+#include <WindowFunctions/IWindowFunction.h>
+#include <WindowFunctions/WindowFunctionFactory.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int BAD_ARGUMENTS;
+extern const int NOT_IMPLEMENTED;
+} // namespace ErrorCodes
+
+struct WindowFunctionRank final : public IWindowFunction
+{
+    WindowFunctionRank(const std::string & name_,
+                       const DataTypes & argument_types_)
+        : IWindowFunction(name_, argument_types_)
+    {}
+
+    DataTypePtr getReturnType() const override
+    {
+        return std::make_shared<DataTypeInt64>();
+    }
+
+    void windowInsertResultInto(WindowBlockInputStreamPtr stream,
```

Comment:
Is it necessary to use `WindowBlockInputStreamPtr` here?
Can we just use `windowInsertResultInto(peer_group_start_row_number, IColumn)`?
such as `ColumnAggregateFunction::insertMergeFrom(const IColumn & src_, size_t n)`
Using `WindowBlockInputStreamPtr` is a bit weird..

Reply:
It is  reserved for  Lead/lag  which use many member variables of WindowBlockInputStream.